### PR TITLE
Flow type electron

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] Flow type some very under typed files
- [x] Flow type electron and introduce SafeElectron

All electron raw imports should use SafeElectron!

### CODE REVIEW

File: shared/app/main-shared.native.js [hunk 1]
```diff
@@ -4,7 +4,6 @@ import * as NotificationsGen from '../actions/notifications-gen'
 import Push from './push/push.native'
 import React, {Component, Fragment} from 'react'
 import RenderRoute from '../route-tree/render-route'
-import loadPerf from '../util/load-perf'
```
Comment (chrisnojima) [inline shared/app/main-shared.native.js]: doesn't really do anything anymore

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

File: shared/actions/fs/platform-specific.desktop.js [hunk 2]
```diff
@@ -183,8 +183,8 @@ function* installFuseSaga(): Saga.SagaGenerator<any, any> {
 }
 
 function uninstallKBFSConfirmSaga(action: FsGen.UninstallKBFSConfirmPayload) {
-  const dialog = Electron.dialog || Electron.remote.dialog
-  dialog.showMessageBox(
+  SafeElectron.getDialog().showMessageBox(
+    undefined,
```
[the remark this answers is not in the document]
Comment (buoyad) [reply shared/actions/fs/platform-specific.desktop.js]: Getting this exception trying to uninstall kbfs via settings: 
```
Error (global): Error: Could not call remote function 'showMessageBox'. Check that the function signature is correct. Underlying error: Error processing argument at index 11, conversion failure from undefined
```

**Pull request merged**